thorclient/httpclient/client.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	var filteredEvents []events.FilteredEvent
+	if err = json.Unmarshal(body, &filteredEvents); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall events - %w", err)
+	}
+
+	return filteredEvents, nil
+}
+
+func (c *Client) FilterTransfers(req *events.EventFilter) ([]*transfers.FilteredTransfer, error) {
+	body, err := c.httpPOST(c.url+"/logs/transfer", req)
+	if err != nil {
+		return nil, fmt.Errorf("unable to send retrieve transfer logs - %w", err)
+	}
+
+	var filteredEvents []*transfers.FilteredTransfer
+	if err = json.Unmarshal(body, &filteredEvents); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall events - %w", err)
+	}
+
+	return filteredEvents, nil
+}
+
+func (c *Client) GetAccount(addr *thor.Address, revision string) (*accounts.Account, error) {
+	url := c.url + "/accounts/" + addr.String()
+	if revision != "" {
+		url += "?revision=" + revision
+	}
+
+	body, err := c.httpGET(url)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve account - %w", err)
+	}
+
+	var account accounts.Account
+	if err = json.Unmarshal(body, &account); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall events - %w", err)
+	}
+
+	return &account, nil
+}
+
+func (c *Client) GetAccountCode(addr *thor.Address, revision string) (*accounts.GetCodeResult, error) {
+	url := c.url + "/accounts/" + addr.String() + "/code"
+	if revision != "" {
+		url += "?revision=" + revision
+	}
+
+	body, err := c.httpGET(url)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve account - %w", err)
+	}
+
+	var res accounts.GetCodeResult
+	if err = json.Unmarshal(body, &res); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall code - %w", err)
+	}
+
+	return &res, nil
+}
+
+func (c *Client) GetAccountStorage(addr *thor.Address, key *thor.Bytes32, revision string) (*accounts.GetStorageResult, error) {
+	url := c.url + "/accounts/" + addr.String() + "/key/" + key.String()
+	if revision != "" {
+		url += "?revision=" + revision
+	}
+
+	body, err := c.httpGET(url)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve account - %w", err)
+	}
+
+	var res accounts.GetStorageResult
+	if err = json.Unmarshal(body, &res); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall code - %w", err)
+	}
+
+	return &res, nil
+}
+
+func (c *Client) GetBlockExpanded(revision string) (*blocks.JSONExpandedBlock, error) {
+	body, err := c.httpGET(c.url + "/blocks/" + revision + "?expanded=true")
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve block - %w", err)
+	}
+
+	var block blocks.JSONExpandedBlock
+	if err = json.Unmarshal(body, &block); err != nil {
+		return nil, fmt.Errorf("unable to unmarshall events - %w", err)
+	}
+
+	return &block, nil
+}
+
+func (c *Client) GetBlock(blockID string) (*blocks.JSONCollapsedBlock, error) {
+	body, err := c.httpGET(c.url + "/blocks/" + blockID)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve block - %w", err)
+	}
+
+	var block blocks.JSONCollapsedBlock
```

Comment:
Handle the scenario that returns `[]byte{"null"}`

Reply:
Fixed 👍